cmd/image/info/info.go

```diff
@@ -0,0 +1,211 @@
+		return fmt.Errorf("%s: illegal file path", filePath)
+	}
+	return nil
+}
+
+// Extract specific file from the tar file
+func Untar(tarball, target, filename string) error {
+	reader, err := os.Open(tarball)
+	if err != nil {
+		return err
+	}
+	defer reader.Close()
+	tarReader := tar.NewReader(reader)
+
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+
+		if header.Name != filename {
+			continue
+		}
+		err = SanitizeExtractPath(header.Name, target)
+		if err != nil {
+			return err
+		}
+		path := filepath.Join(target, header.Name)
+		info := header.FileInfo()
+		if info.IsDir() {
+			if err = os.MkdirAll(path, info.Mode()); err != nil {
+				return err
+			}
+			continue
+		}
+
+		file, err := os.OpenFile(path, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, info.Mode())
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+		_, err = io.Copy(file, tarReader)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+var Cmd = &cobra.Command{
+	Use:   "info",
+	Short: "Provide info about the image",
+	Long: `Provide info about the image
+pvsadm image info --help for information
+
+
+Examples:
+
+# To get the pvsadm tool version by specifying the image
+pvsadm image info --file rhcos-46-12152021.ova.gz
+
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		opt := pkg.ImageCMDOptions
+		var ova string
+		ovaImgDir, err := os.MkdirTemp(opt.TemDir, "ova-img-dir")
+		if err != nil {
+			return err
+		}
+
+		defer os.RemoveAll(ovaImgDir)
+
+		//Check if the image is in gzip format and unzip it.
+		checkGzip, err := isGzip(opt.Filename)
+		if err != nil {
+			return fmt.Errorf("failed to detect the image filetype: %v", err)
+		}
+		if checkGzip {
+			klog.Infof("Image %s is in gzip format, extracting it", opt.Filename)
+			ova = filepath.Join(ovaImgDir, "image.ova")
+			err = gunzipIt(opt.Filename, ova)
+			if err != nil {
+				return err
+			}
+			klog.Infof("Extract complete")
+		} else {
+			ova = opt.Filename
+		}
+
+		//Extract the ovf file
+		err = Untar(ova, ovaImgDir, "coreos.ovf")
+		if err != nil {
+			return err
+		}
+		filename := filepath.Join(ovaImgDir, "coreos.ovf")
+
+		type Ovf struct {
+			Pvsadmversionsection struct {
+				Info        string `xml:"Info"`
+				BuildNumber string `xml:"BuildNumber"`
+			} `xml:"pvsadmversionsection"`
+		}
+
+		xmlFile, err := os.Open(filename)
+		if err != nil {
+			return err
+		}
+		defer xmlFile.Close()
+		var version Ovf
+		byteValue, _ := ioutil.ReadAll(xmlFile)
+		xml.Unmarshal(byteValue, &version)
+		toolversion := version.Pvsadmversionsection.BuildNumber
+		if toolversion == "" {
+			return fmt.Errorf("unable to find the pvsadm tool version")
+		}
+		klog.Infof("Pvsadm tool version used for creating this image: %+v", toolversion)
+		return nil
+	},
+}
+
+// Init method
+func init() {
+	Cmd.Flags().StringVarP(&pkg.ImageCMDOptions.Filename, "file", "f", "", "The PATH to the image")
```

Comment:
instead of --file argument, I'm wondering if we can have something like `pvsadm image info <image.ova>` format for better user experience?

Reply:
Done.